lib/puppetserver/ca/puppet_config.rb

```diff
@@ -0,0 +1,121 @@
+
+module Puppetserver
+  module Ca
+    # Provides an interface for asking for Puppet[ Server] settings w/o loading
+    # either Puppet or Puppet Server. Includes a simple ini parser that will
+    # ignore Puppet's more complicated conventions.
+    class PuppetConfig
+
+      def self.parse(config_path = nil)
+        instance = new(config_path)
+        instance.load
+
+        return instance
+      end
+
+      attr_reader :errors, :settings
+
+      def initialize(supplied_config_path = nil)
+        @using_default_location = !supplied_config_path
+        @config_path = supplied_config_path || user_specific_conf_file
+
+        @settings = nil
+        @errors = []
+      end
+
+      # Return the correct confdir. We check for being root (user id == 0)
+      # on *nix, else the user path. We do not include a check for running
+      # as Adminstrator since non-development scenarios for Puppet Server
+      # on Windows are unsupported.
+      def user_specific_conf_dir
+        if Gem.win_platform? && Process::UID.eid == 0
+          '/etc/puppetlabs/puppet'
+        else
+          "#{ENV['HOME']}/.puppetlabs/etc/puppet"
+        end
+      end
+
+      def user_specific_conf_file
+        user_specific_conf_dir + '/puppet.conf'
+      end
+
+      def load
+        unless @using_default_location && !File.exist?(@config_path)
+          results = parse_text(File.read(@config_path))
+        end
+
+        results ||= {}
+        results[:main] ||= {}
+        results[:master] ||= {}
+
+        overrides = results[:main].merge(results[:master])
+
+        @settings = resolve_settings(overrides).freeze
```

Comment:
Might also be nice to log the final resolved settings at the verbose level. Should probably also log at the various stages so they can track their settings back to where they originate.